### ISSUE TITLE
IDE doesn't suggest upgrading LangVersion when emitting error as warning

### ISSUE THREAD

**Issue by jjonescz** [dotnet/roslyn#69228]
**Version Used**: https://github.com/dotnet/roslyn/pull/69104/commits/e1a748062476da42e0cda2b1532d7c3caff5ee2a

**Steps to Reproduce**:

```csproj
<PropertyGroup>
  <LangVersion>11</LangVersion>
  <GenerateDocumentationFile>true</GenerateDocumentationFile>
</PropertyGroup>
```

```cs
/// <summary>
/// <see cref="M(ref readonly int)"/>
/// </summary>
class C
{
    public static void M(ref readonly int x)
    {
        Console.WriteLine(x);
    }
}
```

**Diagnostic Id**: CS1658, CS8652

**Expected Behavior**: Visual Studio suggests upgrading language version as a code action on the `cref="M(ref readonly int)"` where the following warning is reported:

<samp>CS1658: The feature 'ref readonly parameters' is currently in Preview and *unsupported*. To use Preview features, use the 'preview' language version.. See also error CS8652.</samp>

**Actual Behavior**: Visual Studio suggest upgrading language version only for error CS8652 emitted for `public static void M(ref readonly int x)` declaration, but not when the same error is emitted as warning inside the doc comment (implemented here - notice the `forceWarning: true` parameter):

https://github.com/dotnet/roslyn/blob/e1a748062476da42e0cda2b1532d7c3caff5ee2a/src/Compilers/CSharp/Portable/Binder/Binder_Crefs.cs#L938

Note that this example is about `ref readonly` parameters, but similar issue probably holds for other scenarios.

**Comment by CyrusNajmabadi** [comment]
Closing out as too niche.
